### ISSUE TITLE
Remove upper bound version constraints

### ISSUE THREAD

**Issue by ndawe** [ydataai/ydata-profiling#1293]
### Missing functionality

`ydata-profiling` is difficult to incorporate into an existing stack due to a number of dependency conflicts because of the upper bound version constraints: https://github.com/ydataai/ydata-profiling/blob/master/requirements.txt

Are these upper bound constraints necessary or would you be willing to remove them going forward? https://iscinumpy.dev/post/bound-version-constraints/

### Proposed feature

Upper bound version constraints are removed from `requirements.txt`.

**Comment by aquemy** [comment]
Hi @ndawe,

We had some internal debates about removing the upper bound!
The consensus we reached is to remove it for large libraries with an active release cycle (e.g. pandas, numpy, scipy) and keep the version fully pinned for smaller ones, with less activity. In case we observe problems in the future (as we did in the past!), we will have to pin them back.

This is on the the roadmap for the next version.
